aks-node-controller/app.go

```diff
@@ -92,7 +96,17 @@ func (a *App) Provision(ctx context.Context, flags ProvisionFlags) error {
 
 	config, err := nodeconfigutils.UnmarshalConfigurationV1(inputJSON)
 	if err != nil {
-		return fmt.Errorf("unmarshal provision config: %w", err)
+		// We try our best to continue unmarshal even if there are unexpected situations such as unknown fields.
+		// It usually happens when a newer version of aksNodeConfig is being parsed by an older version of aks-node-controller.
+		// This allows older versions of aks-node-controller to read configurations that may have fields added in newer versions.
+		// Log the error and continue processing.
+		// Note: This may result in loss of data if the unknown fields are critical but that's the best we can do for backward compatibility so far.
+
+		slog.Warn("Unmarshalling aksNodeConfigv1 encounters error but the process will continue."+
```

Comment:
I usually prefer to avoid WARN level and use either INFO or ERROR.
It's unlikely we're going to threat it any different to INFO or ERROR. But it often incorrectly filtered out in log queries.

Reply:
Kinda makes sense. I will use info too.